src/Model/batchprocessing/BatchProcessKaplanMeier.py

```diff
@@ -0,0 +1,163 @@
+import kaplanmeier as km
+import matplotlib as mp1
+mp1.use("TkAgg")
+from matplotlib import pyplot as plt
+import pandas as pd
+from src.Model.batchprocessing.BatchProcess import BatchProcess
+from src.Model.PatientDictContainer import PatientDictContainer
+import logging
+
+
+class BatchProcessKaplanMeier(BatchProcess):
+    """
+    This class handles batch processing for the Clinical Data 2 CSV
+    process. Inherits from the BatchProcessing class.
+    """
+    # Allowed classes for ClinicalDataSR2CSV
+    allowed_classes = {
+        # Comprehensive SR
+        "1.2.840.10008.5.1.4.1.1.88.33": {
+            "name": "sr",
+            "sliceable": False
+        }
+    }
+
+    def __init__(self, progress_callback, interrupt_flag, patient_files, data_Dict,
+                 kaplanmeier_targetCol, kaplanmeier_DurationOfLifeCol, kaplanmeier_AliveOrDeadCol):
+        """
+        Class initialiser function.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param patient_files: List of patient files.
+        :param output_path: Path of the output CSV file.
+        """
+        # Call the parent class
+        super(BatchProcessKaplanMeier, self).__init__(progress_callback,
+                                                             interrupt_flag,
+                                                             patient_files)
+
+        # Set class variables
+        self.patient_dict_container = PatientDictContainer()
+        self.required_classes = ['sr']
+        self.ready = self.load_images(patient_files, self.required_classes)
+        self.kaplanmeier_targetCol = kaplanmeier_targetCol
+        self.kaplanmeier_ = kaplanmeier_targetCol
+        self.kaplanmeier_targetCol = kaplanmeier_targetCol
+        self.kaplanmeier_DurationOfLifeCol = kaplanmeier_DurationOfLifeCol
+        self.kaplanmeier_AliveOrDeadCol = kaplanmeier_AliveOrDeadCol
+        self.dataDict = data_Dict
+
+    def start(self):
+        """
+        Goes through the steps of the ClinicalData-SR2CSV conversion.
+        :return: True if successful, False if not.
+        """
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
```

Comment:
extract this out into a separate method "check_interrupt" or similar. call as need